Assets/Scripts/Systems/Modules/ModuleSlot.cs

```diff
@@ -0,0 +1,26 @@
+using System;
+
+namespace Systems.Modules
+{
+    /// <summary>
+    ///     Represents different module types
+    /// </summary>
+    [Flags]
+    [Serializable]
+    public enum ModuleType
+    {
+        None,
+        Weapon,
+        Engine
+    }
+
+    /// <summary>
+    ///     Stores the valid module types for a specific position on the module grid
+    /// </summary>
+    [Serializable]
+    public class ModuleSlot
+    {
+        public Coordinates Position;
```

Comment:
I would replace this with a reference to the Module currently occupying the slot (and null if empty).